aptos-move/move-examples/marketplace/sources/coin_listing.move

```diff
@@ -0,0 +1,1080 @@
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::object::{Self, ConstructorRef, Object, ObjectCore};
+    use aptos_framework::timestamp;
+
+    use marketplace::fee_schedule::{Self, FeeSchedule};
+    use marketplace::listing::{Self, Listing};
+
+    /// There exists no listing.
+    const ENO_LISTING: u64 = 1;
+    /// This is an auction without buy it now.
+    const ENO_BUY_IT_NOW: u64 = 2;
+    /// The proposed bid is insufficient.
+    const EBID_TOO_LOW: u64 = 3;
+    /// The auction has not yet ended.
+    const EAUCTION_NOT_ENDED: u64 = 4;
+    /// The auction has already ended.
+    const EAUCTION_ENDED: u64 = 5;
+    /// The entity is not the seller.
+    const ENOT_SELLER: u64 = 6;
+
+    // Core data structures
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed-price market place listing.
+    struct FixedPriceListing<phantom CoinType> has key {
+        /// The price to purchase the item up for listing.
+        price: u64,
+        /// Purchase event -- as it is only ever executed once.
+        purchase_event: EventHandle<PurchaseEvent>,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// An auction-based listing with optional buy it now semantics.
+    struct AuctionListing<phantom CoinType> has key {
+        /// Starting bid price.
+        starting_bid: u64,
+        /// Price increment from the current bid.
+        bid_increment: u64,
+        /// Current bid, if one exists.
+        current_bid: Option<Bid<CoinType>>,
+        /// Auction end time in Unix time as seconds.
+        auction_end_time: u64,
+        /// If a bid time comes within this amount of time before the end bid, extend the end bid
+        /// to the current time plus this amount.
+        minimum_bid_time_before_end: u64,
+        /// Buy it now price, ends auction immediately.
+        buy_it_now_price: Option<u64>,
+        /// Bid events
+        bid_events: EventHandle<BidEvent>,
+        /// Purchase event -- as it is only ever executed once.
+        purchase_event: EventHandle<PurchaseEvent>,
+    }
+
+    /// Represents a single bid within this auction house.
+    struct Bid<phantom CoinType> has store {
+        bidder: address,
+        coins: Coin<CoinType>,
+    }
+
+    /// An event triggered upon each bid.
+    struct BidEvent has drop, store {
+        new_bidder: address,
+        new_bid: u64,
+        previous_bidder: Option<address>,
+        previous_bid: Option<u64>,
+    }
+
+    /// An event triggered upon the sale of an item. Note, the amount given to the seller is the
+    /// price - commission - royalties. In the case there was no sale, purchaser is equal to seller
+    /// and the amounts will all be zero.
+    struct PurchaseEvent has drop, store {
+        purchaser: address,
+        price: u64,
+        commission: u64,
+        royalties: u64,
+    }
+
+    // Init functions
+
+    #[legacy_entry_fun]
```

Comment:
Although this does allow the entry fun to return a value, did you check that this entry function is still callable as a tx? Also, it's not good to encourage using this attribute as it also skips all args checking during compilation, not just return value